index.src.html

```diff
@@ -2289,6 +2301,13 @@
             |clone|.
     2. Return |clone|.
 
+: To check if a {{AudioDataInit}} is a
+    <dfn>valid AudioDataInit</dfn>, run these steps:
+:: 1. If {{AudioDataInit/sampleRate}} = 0, return `false`.
+    2. If {{AudioDataInit/numberOfFrames}} = 0, return `false`.
+    3. If {{AudioDataInit/numberOfChannels}} = 0, return `false`.
```

Comment:
Should we go further and check that |data| has enough bytes to cover `frames x channels x bytesPerFrame`?

Reply:
Yes. I wouldn't limit the input length, but copyTo() shouldn't be able to copy more than that many bytes either.

Reply:
I've added a check. Yes, not limiting the input seems good, although the spec does say we are making a copy here. Maybe we should copy only the right length.

Reply:
I think what you've got so far is good. AFAICT, only copying the right length is not actually observable. We already ensure that later calls to copyTo() are bounded by the frame count.

Reply:
I've added steps to reject the AudioDataInit if the `data` provided doesn't cover the total number of samples.